plugin/src/main/java/io/jenkins/plugins/casc/core/JenkinsConfigurator.java

```diff
@@ -97,9 +104,39 @@ protected Jenkins instance(Mapping mapping, ConfigurationContext context) {
                 .getter(j -> j.proxy)
                 .setter((o, v) -> o.proxy = v));
 
+        attributes.add(new MultivaluedAttribute<Jenkins, NodeMonitor>("nodeMonitors", NodeMonitor.class)
+                .getter(j -> ComputerSet.getMonitors())
+                .setter((jenkins, nodeMonitors) -> {
+                    DescribableList<NodeMonitor, Descriptor<NodeMonitor>> monitors = ComputerSet.getMonitors();
+                    monitors.clear();
+                    monitors.addAll(nodeMonitors);
+                    for (Descriptor<NodeMonitor> d : NodeMonitor.all()) {
+                        if (monitors.get(d) == null) {
+                            NodeMonitor i = createDefaultInstance(d);
```

Comment:
why does this need doing?

Reply:
By default all node monitors are somehow active and show data, but when a nodemonitor is ignored (i.e. on the config page they are not checked), it should mean the monitor should not take agents offline when the monitor detects a problem.
If I do not add the monitor here it would not be shown at all on .../computer, but on the configure page they would be visible and after saving also shown.
This logic is copied from ComputerSet class